### PR TITLE
Ports server news from Polaris.

### PR DESCRIPTION
**Pull request by ike709**
:cl: ike709
add: Ported the Server News window from Polaris so players have another thing to ignore.
/:cl:

I will go insane if I don't do something besides bugfixes and mirror PRs.

Ports https://github.com/PolarisSS13/Polaris/pull/2909

Don't ask me why the new_player.dm copypasta is a thing. Fixing that is out of the scope of this PR.

### CODE REVIEW

File: code/modules/admin/news.dm [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#define NEWSFILE "data/news.sav"	//where the memos are saved
+
+/client/
+	var/last_news_hash = null // Stores a hash of the last news window it saw, which gets compared to the current one to see if it is different.
+
+// Returns true if news was updated since last seen.
+/client/proc/check_for_new_server_news()
+	var/savefile/F = get_server_news()
+	if(F)
+		if(md5(F["body"]) != last_news_hash)
```
Comment (ninjanomnom) [inline code/modules/admin/news.dm]: Shorten this to `if(F && md5(F["body"]) != last_news_hash)`

File: code/modules/admin/news.dm [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+			return TRUE
+	return FALSE
+
+/client/proc/modify_server_news()
+	set name = "Modify Public News"
+	set category = "Server"
+
+	if(!check_rights(0))
```
Comment (ninjanomnom) [inline code/modules/admin/news.dm]: This should probably be `if(!check_rights(R_SERVER))`

File: code/modules/admin/news.dm [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		return
+
+	var/savefile/F = new(NEWSFILE)
+	if(F)
+		var/title = F["title"]
+		var/body = F["body"]
+		var/new_title = copytext(sanitize(input(src,"Write a good title for the news update.  Note: HTML is NOT supported.","Write News", title) as null|text),1,MAX_MESSAGE_LEN)
+		if(!new_title || findtext(new_title,"<script",1,0)) // Is this needed with sanitize()?)
+			return
```
Comment (ninjanomnom) [inline code/modules/admin/news.dm]: This is probably unnecessary as the comment suggests but someone will need to investigate sanitize. I don't know it off hand.

File: code/modules/admin/news.dm [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+		var/new_body = copytext(sanitize(input(src,"Write the body of the news update here. Note: HTML is NOT supported.","Write News", body) as null|message),1,MAX_MESSAGE_LEN)
+		if(!new_body || findtext(new_body,"<script",1,0))
+			return
```
Comment (ninjanomnom) [inline code/modules/admin/news.dm]: See above.

File: code/modules/admin/news.dm (hunk 3 above)
Comment (ike709) [inline code/modules/admin/news.dm]: Well the button is on the server tab. Anyone with access to that can fuck everything up in worse ways. So I'll just remove it.

**Comment by ike709** [comment]
wew lad
